Content.Server/Lathe/LatheSystem.cs

```diff
@@ -73,14 +73,20 @@ private void OnLatheEjectMessage(EntityUid uid, LatheComponent lathe, LatheEject
                     return;
 
                 var volumePerSheet = composition.MaterialComposition.FirstOrDefault(kvp => kvp.Key == message.Material).Value;
-                int sheetsToExtract = Math.Min(message.SheetsToExtract, _stack.GetMaxCount(material.StackEntity));
+                var sheetsToExtract = Math.Min(message.SheetsToExtract, _stack.GetMaxCount(material.StackEntity));
 
                 volume = sheetsToExtract * volumePerSheet;
             }
 
             if (volume > 0 && _materialStorage.TryChangeMaterialAmount(uid, message.Material, -volume))
             {
-                _materialStorage.SpawnMultipleFromMaterial(volume, material, Transform(uid).Coordinates, out var overflow);
+                var mats = _materialStorage.SpawnMultipleFromMaterial(volume, material, Transform(uid).Coordinates, out _);
+                foreach (var mat in mats)
+                {
+                    if (TerminatingOrDeleted(mat))
+                        continue;
+                    _stack.TryMergeToContacts(mat);
+                }
             }
```

Comment:
I don't have my IDE open but this doesn't spawn an entity every unit right.

Reply:
nah this basically just spawns X stacks of Y count to fulfill Z amount of materials.

i.e. if i want to print 500 steel material, it will spawn one steel sheet entity with a count of 5.